### PR TITLE
Nikos/5096/improve the integration test for contracts default attributes

### PR DESCRIPTION
**Pull request by nikoulai**
## Description

#5096 

## Checklist for 4.x:

- [ ] I have selected the correct 4.x base branch.
- [ ] Within the description, the feature or issue is discussed in detail or an issue is linked.   
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have added any required tests for the changes I made 
- [ ] I have made corresponding changes to the documentation
- [ ] Any dependent changes have been merged and published in downstream modules.
- [ ] I ran `yarn` successfully
- [ ] I ran `yarn lint` successfully
- [ ] I ran `yarn build:web` successfully
- [ ] I ran `yarn test:unit` successfully
- [ ] I ran `yarn test:integration` successfully
- [ ] I ran `compile:contracts` successfully
- [ ] I have tested my code.
- [ ] I have updated the corresponding `CHANGELOG.md` file in the packages I have edited.

### CODE REVIEW

File: packages/web3-eth-contract/test/integration/contract_defaults_extra.test.ts [hunk 1]
```diff
@@ -0,0 +1,390 @@
...
+		sendOptions = { from: acc.address, gas: '1000000' };
+	});
+
+	describe('defaultHardfork', () => {
+		it('should use "defaultHardfork" on "Contract" level', async () => {
+			const hardfork = 'berlin';
+
+			Contract.defaultHardfork = hardfork;
+
+			contract = await contract.deploy(deployOptions).send(sendOptions);
```
Comment (jdevcs) [inline packages/web3-eth-contract/test/integration/contract_defaults_extra.test.ts:80]: Tx passed to https://github.com/web3/web3.js/blob/a82f589f05ec3b427547e0440607e89c8962655f/packages/web3-eth-contract/src/contract.ts#L1137 should also be checked for hardfork value

File: packages/web3-eth-contract/test/integration/contract_defaults_extra.test.ts [hunk 2]
```diff
@@ -0,0 +1,390 @@
...
+				data: GreeterBytecode,
+				arguments: ['My Greeting'],
+			};
+
+			sendOptions = { from: acc.address, gas: '1000000' };
+
+			contract = await contract.deploy(deployOptions).send(sendOptions);
+		});
+
+		// todo this test fails, seems like a bug. any thoughts?
```
Comment (jdevcs) [inline packages/web3-eth-contract/test/integration/contract_defaults_extra.test.ts:182]: User should be able to set and get `defaultCommon` and it should be used before local tx signing ( via eth pkg), Is it failing at line 192?, if this tests if failing its bug.

**Review by jdevcs — CHANGES_REQUESTED** [review]

**Comment by jdevcs** [comment]
Good work

File: packages/web3-eth-contract/test/integration/contract_defaults_extra.test.ts (hunk 2 above)
Comment (nikoulai) [reply packages/web3-eth-contract/test/integration/contract_defaults_extra.test.ts:182]: yes, it's failing

File: packages/web3-eth-contract/test/integration/contract_defaults_extra.test.ts (hunk 1 above)
Comment (jdevcs) [reply packages/web3-eth-contract/test/integration/contract_defaults_extra.test.ts:80]: For failing tests could you add `skip` to test and create a new issue for bug investigation.

File: packages/web3-eth-contract/package.json [hunk 3]
```diff
@@ -49,6 +49,7 @@
 		"jest-extended": "^3.0.1",
 		"prettier": "^2.7.1",
 		"ts-jest": "^28.0.7",
-		"typescript": "^4.7.4"
+		"typescript": "^4.7.4",
+		"web3-eth-accounts": "^4.0.1-alpha.0"
```
Comment (luu-alex) [inline packages/web3-eth-contract/package.json:53]: this change looks unneccesary

Comment (nikoulai) [reply packages/web3-eth-contract/package.json:53]: Do you mean typescript or accounts change? For the former, a comma was added, and for the latter I need the package for testing.

Comment (luu-alex) [reply packages/web3-eth-contract/package.json:53]: i am talking about `web3-eth-accounts` I think if its only for testing `web3-eth-accounts` should be under dev dependancies and it [seems like it is](https://github.com/web3/web3.js/blob/e8a831879d73024b3aa0ade39f3e93270582eb43/packages/web3-eth-contract/package.json#L53), or am i missing something

Comment (luu-alex) [reply packages/web3-eth-contract/package.json:53]: oh never mind i get it, your right

File: packages/web3-eth-contract/test/integration/contract_defaults_extra.test.ts (hunk 1 above)
Comment (nikoulai) [reply packages/web3-eth-contract/test/integration/contract_defaults_extra.test.ts:80]: Just opened one
https://github.com/web3/web3.js/issues/5603

**Review by jdevcs — APPROVED** [review]

**Review by luu-alex — APPROVED** [review]

**Review by avkos — APPROVED** [review]

**Pull request merged**